### PR TITLE
Search and display a guider's groups

### PR DESCRIPTION
**Pull request by benlovell**
Displays group membership alongside the guider details. Also adds
searchability to the groups fields for guiders.

### CODE REVIEW

**Comment by benbarnett** [comment]
@benlovell the `.on(event, callback)` thing in the [list.js docs](http://www.listjs.com/docs/list-api) might be a useful hook for the test stuff?

File: app/models/user.rb [hunk 1]
```diff
@@ -6,6 +6,8 @@ class User < ApplicationRecord
 
   has_many :schedules, dependent: :destroy
 
+  has_and_belongs_to_many :groups # rubocop:disable Rails/HasAndBelongsToMany
```
Comment (Guntrisoft) [inline app/models/user.rb]: So whats the lowdown on why this needs to be disabled?

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 RSpec.configure do |config|
+  config.order = :random
```
Comment (AndrewVos) [inline spec/spec_helper.rb]: Yes!

File: app/models/user.rb (hunk 1 above)
Comment (benlovell) [inline app/models/user.rb]: Rubocop wants you to use a `has_many :through` instead but this only makes sense when you want / care about attributes on the associative model, and we don't need those.

**Pull request merged**